### PR TITLE
Clean up linking docs re: universal links

### PR DESCRIPTION
**Pull request by bradjones1**
# Why

There is a dead/outdated link in the universal linking docs, as well as a dead link to `example.com` due to auto-linking of URLs. Also adds a note regarding the bare workflow.

# How

Fixes my initial confusion and frustration around handling linking in the bare workflow.

# Test Plan

Manual review.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).

### CODE REVIEW

File: docs/pages/guides/linking.md [hunk 1]
```diff
@@ -247,6 +247,9 @@ Another example of using `WebBrowser` for authentication can be found at [exampl
 
 It is often desirable for regular HTTPS links (without a custom URL scheme) to directly open your app on mobile devices. This allows you to send notification emails with links that work as expected in a web browser on desktop, while opening the content in your app on mobile. iOS refers to this concept as "universal links" while Android calls it "deep links" (but in this section, we are specifically discussing deep links that do not use a custom URL scheme).
 
+> For apps using the Bare Workflow, see the notes on the relevant [configuration keys](../versions/latest/config/app)
```
Comment (Simek) [inline docs/pages/guides/linking.md:250]:
```suggestion
> For apps using the Bare Workflow, see the notes on the relevant [configuration keys](/versions/latest/config/app)
```

Let's not use the relative links paths. 🙂

File: docs/pages/guides/linking.md [hunk 2]
```diff
@@ -331,7 +334,7 @@ At this point, opening a link on your mobile device should now open your app! If
 
 ### Deep links on Android
 
-Implementing deep links on Android (without a custom URL scheme) is somewhat simpler than on iOS. You simply need to add `intentFilters` to the [Android section](/workflow/configuration.md#android) of your **app.json**. The following basic configuration will cause your app to be presented in the standard Android dialog as an option for handling any record links to `myapp.io`:
+Implementing deep links on Android (without a custom URL scheme) is somewhat simpler than on iOS. You simply need to add `intentFilters` to the [Android section](../versions/latest/config/app/#android) of your **app.json**. The following basic configuration will cause your app to be presented in the standard Android dialog as an option for handling any record links to `myapp.io`:
```
Comment (Simek) [inline docs/pages/guides/linking.md:337]:
```suggestion
Implementing deep links on Android (without a custom URL scheme) is somewhat simpler than on iOS. You simply need to add `intentFilters` to the [Android section](/versions/latest/config/app/#android) of your **app.json**. The following basic configuration will cause your app to be presented in the standard Android dialog as an option for handling any record links to `myapp.io`:
```

Same as above.

**Review by Simek — CHANGES_REQUESTED** [review]
Hello @bradjones1, thank you for the updates! 👍 

I have left suggestion according to link paths, but besides that this LGTM!

**Comment by bradjones1** [comment]
Thanks. I honestly didn't know what was correct and saw both throughout the codebase (and couldn't easily find a note in the contributing doc on what to do?) so that's all fine by me.

**Review by Simek — APPROVED** [review]
Thank you for the updates! 👍 

According to link variants: generally, both approaches are correct, but relative links are more error prone

Also thank you for the hint related to docs Contributing guide! We probably going to expand it in near future, and I will ensure that the docs part includes linking guide/tips. 🙂

**Comment by bradjones1** [comment]
@Simek if you're taking input on the contribution guide, maybe also take a look at https://github.com/expo/expo/issues/16861#issuecomment-1087557473 re: confusion about how to contribute fixes to the configuration schema (it's not in the repos.)

**Pull request merged**